source/core/slang-shared-library.h

```diff
@@ -94,6 +94,13 @@ class TemporarySharedLibrary : public DefaultSharedLibrary
     String m_path;
 };
 
+class SharedLibraryUtils
+{
+public:
+    static String getSharedLibraryFileName(void* symbolInLib);
+    static uint64_t getSharedLibraryTimestamp(void* symbolInLib);
```

Comment:
Putting the getting at time stamp from a file into the slang-io.h/cpp (say into Path). And having typedef def the timstamp type might be nice.